packages/fxa-settings/src/components/Modal/index.tsx

```diff
@@ -2,39 +2,61 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React, { ReactNode } from 'react';
+import React, { ReactNode, useEffect, useRef } from 'react';
 import { useClickOutsideEffect } from '@fxa-react/lib/hooks';
 import classNames from 'classnames';
 import Portal from '@fxa-react/components/Portal';
 import { ReactComponent as CloseIcon } from '@fxa-react/images/close.svg';
 
 type ModalProps = {
   className?: string;
-  onDismiss?: Function;
+  onDismiss: Function;
   onConfirm?: () => void;
   children: ReactNode;
+  headerId: string;
+  descId: string;
   'data-testid'?: string;
 };
 
-/* istanbul ignore next - not worth testing this function */
-const noop = () => {};
-
 const Modal = ({
   className = '',
   onDismiss,
   onConfirm,
   children,
+  headerId,
+  descId,
   'data-testid': testid = 'modal',
 }: ModalProps) => {
-  const modalInsideRef = useClickOutsideEffect<HTMLDivElement>(
-    // HACK: can't use the hook conditionally, so let's supply a dummy
-    // function when onDismiss is missing
-    onDismiss || noop
-  );
+  const modalInsideRef = useClickOutsideEffect<HTMLDivElement>(onDismiss);
+
+  // direct tab focus to the modal when opened for screenreaders
+  const tabFenceRef = useRef<HTMLDivElement>(null);
+  useEffect(() => {
+    if (tabFenceRef.current) {
+      tabFenceRef.current.focus();
+    }
+  }, []);
+
+  // close on esc keydown
+  useEffect(() => {
+    const handler = ({ key }: KeyboardEvent) => {
+      if (key === 'Escape') {
```

Comment:
Is `Escape` reliable here? Or do we need to check a KeyCode?

Reply:
Yeah good call out, when I actually made this I wondered why 'Escape' worked so effortlessly, but I did look up examples doing this and they don't use a key code. I'll double check this will be reliable.

Reply:
So it turns out `key` [has wide support now](https://caniuse.com/#search=KeyboardEvent.key). We don't support IE11 so this should be fine. (`keyCode` is [actually deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode))

Reply:
Hooray!